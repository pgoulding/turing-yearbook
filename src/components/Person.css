.User__card{
  background-color: rgb(222, 239, 253);
  border-radius: 30px;
  box-sizing: border-box;
  padding: 5%;
  opacity: .97;
}

h3 {
  color:red;
}

.User__quote{
  font-style: italic;
}

.User__superlative{
  color: dodgerblue;
  font-weight: 900;
}

img.User__profile-photo {
  width: 100%;
  height:300px;
  border-radius: 30px;
  object-fit:cover;
  opacity: 1.0;
}