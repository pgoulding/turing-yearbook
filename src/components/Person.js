import React, { Component } from 'react';
import './Person.css';
import PropTypes from "prop-types";

// PERSON COMPONENT CODE GOES HERE
class Person extends Component {
  render() {
    return (
      <div className="Instructor__card" data-id={this.props.id}>
        <img src={this.props.image} alt={this.props.name}></img>
        <h3>{this.props.name}</h3>
        <p className="User__quote">{this.props.quote}</p>
        <p>{this.props.superlative}</p>
      </div>
    )
  }
}

export default Person;
