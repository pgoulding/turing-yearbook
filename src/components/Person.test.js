import React from 'react'
import { shallow, mount } from 'enzyme'
import Person from './Person'

describe('Person',() => {
  let wrapper;
  let mockPerson;

  beforeEach(() => {
    mockPerson = { 
      id:1, 
      name:'Bobo', 
      photo:'./test.jpg',
      quote:' I didn\'t do it',
      superlative:'Probably did it' 
    }

    wrapper = shallow(
    <Person person = { mockPerson }/>)
  })

  it('should render a person', () => {
    expect(wrapper).toMatchSnapshot()
  })


})