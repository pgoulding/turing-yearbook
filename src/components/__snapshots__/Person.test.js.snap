// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should render a person 1`] = `
<div
  className="User__card"
  data-id={1}
>
  <img
    alt="Bobo"
    className="User__profile-photo"
    src="./test.jpg"
  />
  <h3>
    Bobo
  </h3>
  <p
    className="User__quote"
  >
     I didn't do it
  </p>
  <p
    className="User__superlative"
  >
    Probably did it
  </p>
</div>
`;
